tests/test_views_phone.py

```diff
@@ -182,7 +183,21 @@ def test_verify(self):
             with self.settings(TWO_FACTOR_TOTP_DIGITS=no_digits):
                 device = PhoneDevice(key=random_hex())
                 self.assertFalse(device.verify_token(-1))
+                time.sleep(1)
                 self.assertFalse(device.verify_token('foobar'))
+                time.sleep(2)  # 4, 8, 16, ..., 2^n
```

Comment:
Introducing sleeps in the tests makes the test suite run slower. I would suggest finding an alternative. For example by using a fake clock that can be forwarded programmatically.